### PR TITLE
Adds telemetry fields specific to datadog traces.

### PR DESCRIPTION
**Pull request by scirner22**

### CODE REVIEW

**Review by zhongzc — APPROVED** [review]
LGTM

Need to run `cargo fmt`.

File: crates/minitrace-datadog/src/lib.rs [hunk 1]
```diff
@@ -9,6 +9,9 @@ use std::net::SocketAddr;
 
 pub fn encode(
     service_name: &str,
+    r#type: &str,
```
Comment (andylokandy) [inline crates/minitrace-datadog/src/lib.rs:12]: Could we pick a normal name for it?

Comment (scirner22) [reply crates/minitrace-datadog/src/lib.rs:12]: Sure. `trace_type`? Do you prefer using serde rename on the struct or keep that `r#type`?

Comment (andylokandy) [reply crates/minitrace-datadog/src/lib.rs:12]: I prefer to use serde rename

**Review by andylokandy — APPROVED** [review]

**Pull request merged**